cmd/frontend/internal/search/search.go

```
@@ -433,13 +434,27 @@ func fromContentMatch(fm *result.FileMatch, repoCache map[api.RepoID]*types.Sear
 		})
 	}
 
+	ctx, cancel := context.WithTimeout(context.TODO(), 5*time.Second) // TODO: context propagation. Discuss: should we propagate parent context to conversion?
```

Why wouldn't we propagate parent context? We can always set a more-strict timeout here, but we should cancel immediately if the parent cancels IMO.

---

Oh wait. I see. There is no parent context immediately available

---

We should propagate parent context--

The part I don't like is that `fromContentMatch` conversion would always succeed, and propagating here would make this conversion function "maybe succeed", when all that I really want to "maybe succeed" is the decoration, not the conversion. 

Teasing out the decoration to a separate function is kind of extra work, and potentially trick (because we want to act on `fm`, but that means we need to do maybe-successful-decoration before calling this conversion function--I don't want to act on the event result type).

We don't have to do it now, I just think logically separating these is worth it. Maybe for now I'll just propagate ctx.

---

I think it's fine to swallow any decoration errors, and just use context propagation for early cancellation

---

Decoration is already a "maybe succeed," so a context error won't change that